osf/utils/storage.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import absolute_import
+
+from django.db import models
+from website.util import api_v2_url
+
+from django.core.files.storage import Storage
+from django.core.files.base import ContentFile
+from django.utils.deconstruct import deconstructible
+
+
+class BannerImage(models.Model):
+    filename = models.TextField(unique=True)
+    image = models.BinaryField()
+
+@deconstructible
+class BannerImageStorage(Storage):
+    def _open(self, name, mode='rb'):
+        assert mode == 'rb'
+        icon = BannerImage.objects.get(filename=name)
+        return ContentFile(icon.image)
+
+    def _save(self, name, content):
+        BannerImage.objects.update_or_create(filename=name, defaults={'image': content.read()})
+        return name
+
+    def delete(self, name):
+        BannerImage.objects.get(filename=name).delete()
+
+    def get_available_name(self, name, max_length=None):
```

Comment:
Get the default `get_available_name` looks for available names on the os, but these files are being stored in the database.

 A thing to note here is that if you name two images the same thing the older one will be overwritten. It could be a good move to make this more like the original `get_available_name` which appends random numbers to the end of filenames if there are collisions. (Should I add this?) One reason not to would be that if we happen to use the same mobile banner for a whole campaign we wouldn't want it stored 5x (but that is unlikely).